contracts/governance/Staking/modules/StakingWithdrawModule.sol

```diff
@@ -250,7 +251,10 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
         bool isGovernance
     ) internal {
         if (_isVestingContract(msg.sender)) {
-            uint256 nextLock = until.add(TWO_WEEKS);
+            // adjust nextLock to the next valid lock date to make sure we don't accidentally
+            // withdraw stakes that are in the future and would get slashed (if until is not
+            // a valid lock date)
+            uint256 nextLock = _adjustDateForOrigin(until.add(TWO_WEEKS));
```

Comment:
this is correct, but we could save some gas by adjusting `until` in the public function `withdraw` , so we do not have to adjust 2 dates but just 1 (if until is already adjusted, we know that until + 2 weeks is valid.